### ISSUE TITLE
iPython hangs when calling show with Matplotlib 1.5.3 and 'TkAgg' backend

### ISSUE THREAD

**Issue by djkirkham** [matplotlib/matplotlib#8845]
### Bug report

**Bug summary**

When using Matplotlib 1.5.3 in iPython with the TkAgg backend, control is not returned to iPython after showing a plot and closing the window.

**Code for reproduction**
```python
In [1]: import matplotlib as mpl

In [2]: mpl.use('TkAgg')

In [3]: import matplotlib.pyplot as plt

In [4]: plt.plot([0, 1])
Out[4]: [<matplotlib.lines.Line2D at 0x7fb09f2a0350>]

In [5]: plt.show()
```

**Actual outcome**

Closing the window does not return control to iPython

**Matplotlib version**
  * Operating System: Linux
  * Matplotlib Version: 1.5.3
  * Python Version: 2.7
  * Other Libraries: iPython 4.1.2, but it happens with other versions too.

Installed via conda via conda-forge channel

The issue doesn't occur in 2.0. Are there likely to be any more 1.5 point releases?

**Comment by dstansby** [comment]
Unfortunately there won't be any more 1.5 point releases, so upgrading to 2.0 is the best option. (is there a reason you're still using 1.5 and not 2.0?)

**Comment by jenshnielsen** [comment]
In IPython it is generally recommended to use the `%matplotlib` magic such as `%matplotlib tk` if you are stuck on 1.5.3 I would recommend trying that

**Comment by tacaswell** [comment]
calling `plt.ion()` should also fix the problem (by making `plt.show()` non-blocking).

Closing as there will be no more 1.5.x releases and is reportedly fixed in 2.0
